### DISCUSSION TITLE
How to read Env variable in custom SPI

### DISCUSSION THREAD

**Discussion by kabipatt** [dasniko/keycloak-extensions-demo#14] [General]
Hi Nico,
Thanks for opening this KC extension code base to the world. Its very  helpful for KC newbies like me. I have a technical queries on your email extension. You mentioned these two ENV variable in your readme.
```
KC_SPI_EMAIL_TEMPLATE_PROVIDER=json
KC_SPI_EMAIL_SENDER_PROVIDER=aws-ses 
```
Are you using this anywhere? The values are same as the PROVIDER_ID set in the factory class.  

I am asking as I am struggling to read an env variable inside my SPI. Here is my example.

1. env Variable is ENV `--spi-my-spi-ruleSet="8443:sso.mypoc.com,80:sso.mypoc.com"` 
2. (OR `KC_SPI_MY_SPI_RULESET="8443:sso.mypoc.com,80:sso.mypoc.com"`
3. My Provider id is "`my-spi`".
4. I am trying to read the variable **ruleSet** by following code, but it is not reading the variable.

```
public class MySpiFactory implements HostnameProviderFactory {
:
    public void init(Config.Scope config) {
	Set<String> props = config.getPropertyNames();
	String ruleSet = config.get("ruleSet");
:
```

**Comment by dasniko** [comment]
https://www.keycloak.org/server/configuration-provider
